src/components/JfListMakerComponent.spec.js

```diff
@@ -1,50 +1,389 @@
-            propsData: {},
-            stubs:["jf-field"]
-        });
-        expect(
-            wrapper.find('.icon.icon-new').attributes('disabled')
-        ).toBe('disabled')
+        it('should display passed in values in the list', function () {
+            const wrapper = mountComponent({
+                value: ['a'],
+            })
+            //text
+            expect(wrapper.find(SELECTOR_A_LIST_ROW).text()).toEqual("a")
+            //delete button
+            expect(
+                wrapper
+                    .contains(
+                        BUTTON_DELETE
+                    )
+            ).toBe(true)
+        })
+
+        it('should display new values in the list', function () {
+
+            const wrapper = mountComponent({
+                value: ['a'],
+            });
+
+            //Set some data
+            wrapper.find(INPUT_TEXT).setValue('b');
+            //Click on the + button
+            wrapper.find(BUTTON_ADD).trigger("click");
+            expect(wrapper.findAll(SELECTOR_A_LIST_ROW).at(0).text()).toEqual("a")
+            expect(wrapper.findAll(SELECTOR_A_LIST_ROW).at(1).text()).toEqual("b")
+        })
+
+        it('should display new values in the list', function () {
+            const wrapper = mountComponent({
+                value: ["a", "b"],
+            });
+            expect(wrapper.findAll(SELECTOR_A_LIST_ROW).at(0).text()).toEqual("a")
+            expect(wrapper.findAll(SELECTOR_A_LIST_ROW).at(1).text()).toEqual("b")
+        })
     })
-    it('The + button should be enabled when any text is typed in', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs:["jf-field"]
-        });
-        wrapper.setData({ newValue: 'a' })
-        expect(
-            wrapper.find('.icon.icon-new').attributes('disabled')
-        ).not.toBe('disabled')
+
+
+    describe("events and validations", function() {
+        describe("addition", function() {
+
+            it(`if valid text is entered in the textbox and the + button clicked, it should\n\temit "${EVENT_ON_ADD_VALUE}", "${EVENT_ON_AFTER_ADD_VALUE}" & "${EVENT_INPUT}"`, function() {
+                const wrapper = mountComponent({
+                    value: ['a'],
+                })
+
+                //Set some data
+                wrapper.find(INPUT_TEXT).setValue("b");
+                //Click on the + button
+                wrapper.find(BUTTON_ADD).trigger('click')
+                //One event contains the new value entered
+                expect(
+                    wrapper.emitted()[EVENT_ON_ADD_VALUE][0]
+                ).toEqual([{ newValue: 'b' }])
+                // One event contains the updated array (necessary for two-way binding)
+                expect(wrapper.emitted()[EVENT_INPUT][0]).toEqual([
+                    ['a', 'b'],
+                ])
+                expect(
+                    wrapper.emitted()[EVENT_ON_AFTER_ADD_VALUE]
```

Comment:
Is the 'input' event bound to some method in the component? If yes, you can mock this method with `jest.fn()` and then you can expect if method called and sample the args.

Reply:
The test is validating if component emits the 'input' event.